src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -81,10 +83,36 @@ public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
+    public Answers getAnswers() {
         return answers;
     }
 
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
+        validateOwnership(loginUser);
+
+        answers.validateOwnership(loginUser);
+
+        this.deleted = true;
+
+        List<DeleteHistory> deletedHistories = addQuestionAndAnswersToDeleteHistory(loginUser);
+
+        return deletedHistories;
+    }
+
+    private void validateOwnership(NsUser loginUser) throws CannotDeleteException {
+        if (!isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+    }
+
+    private List<DeleteHistory> addQuestionAndAnswersToDeleteHistory(NsUser loginUser) throws CannotDeleteException {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, this.id, loginUser, LocalDateTime.now()));
+        deleteHistories.addAll(answers.deleteAll());
```

Comment:
`answers.validateOwnership(loginUser)`를 외부에서 먼저 호출한 뒤, `answers.deleteAll()`을 호출하고 있는데
이 로직을 `answers.deleteAll(loginUser)`으로 합쳐 내부에서 소유자 검증까지 함께 처리하면 
더 간결해질 수 있을 것 같네요 😄 

```suggestion
        deleteHistories.addAll(answers.deleteAll(loginUser));
```

Reply:
fc15cdcdf077b76da264a8d9afbd4f2d79fec1ab